packages/components/pie-checkbox-group/src/index.ts

```diff
@@ -1,20 +1,97 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-import { RtlMixin, defineCustomElement } from '@justeattakeaway/pie-webc-core';
-
+import {
+    LitElement, html, unsafeCSS, nothing, PropertyValues,
+} from 'lit';
+import { property } from 'lit/decorators.js';
+import {
+    RtlMixin, defineCustomElement, FormControlMixin, validPropertyValues,
+} from '@justeattakeaway/pie-webc-core';
+import { ifDefined } from 'lit/directives/if-defined.js';
 import styles from './checkbox-group.scss?inline';
-import { CheckboxGroupProps } from './defs';
+import { CheckboxGroupProps, defaultProps, statusTypes } from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-checkbox-group';
+const assistiveTextIdValue = 'assistive-text';
 
 /**
  * @tagname pie-checkbox-group
  */
-export class PieCheckboxGroup extends RtlMixin(LitElement) implements CheckboxGroupProps {
+export class PieCheckboxGroup extends FormControlMixin(RtlMixin(LitElement)) implements CheckboxGroupProps {
+    @property({ type: String })
+    public label?: CheckboxGroupProps['label'];
+
+    @property({ type: String })
+    public assistiveText?: CheckboxGroupProps['assistiveText'];
+
+    @property({ type: String })
+    @validPropertyValues(componentSelector, statusTypes, defaultProps.status)
+    public status = defaultProps.status;
+
+    @property({ type: Boolean, reflect: true })
+    public disabled?: CheckboxGroupProps['disabled'];
+
+    @property({ type: Object })
+    public aria: CheckboxGroupProps['aria'];
+
+    get _slottedChildren () {
+        const slot = this.shadowRoot?.querySelector('slot');
+        return slot?.assignedElements();
+    }
+
+    private _handleDisabled () : void {
+        if (this._slottedChildren) {
+            [...this._slottedChildren]
+            .forEach((child) => child.setAttribute('disabled', 'true'));
+        }
+    }
+
+    protected firstUpdated (_changedProperties: PropertyValues<this>): void {
```

Comment:
I think `updated` is called immediately after `firstUpdated`, so I wonder if we need both?